src/tile.cpp

```diff
@@ -396,94 +394,96 @@ void Tile::onUpdateTileItem(Item* oldItem, const ItemType& oldType, Item* newIte
 			}
 		}
 	} else if (oldItem->hasProperty(CONST_PROP_MOVEABLE) || oldItem->getContainer()) {
-		auto it = g_game.browseFields.find(this);
+		auto it = g_game.browseFields.find(getTile());
 		if (it != g_game.browseFields.end()) {
-			Cylinder* oldParent = oldItem->getParent();
+			auto oldParent = oldItem->getParent();
 			it->second->removeThing(oldItem, oldItem->getItemCount());
 			oldItem->setParent(oldParent);
 		}
 	}
 
-	const Position& cylinderMapPos = getPosition();
+	const Position& thingMapPos = getPosition();
 
 	SpectatorVec spectators;
-	g_game.map.getSpectators(spectators, cylinderMapPos, true);
+	g_game.map.getSpectators(spectators, thingMapPos, true);
 
 	// send to client
-	for (Creature* spectator : spectators) {
-		if (Player* spectatorPlayer = spectator->getPlayer()) {
-			spectatorPlayer->sendUpdateTileItem(this, cylinderMapPos, newItem);
+	for (const auto& spectator : spectators) {
+		if (auto spectatorPlayer = spectator->getPlayer()) {
+			spectatorPlayer->sendUpdateTileItem(getTile(), thingMapPos, newItem);
 		}
 	}
 
 	// event methods
-	for (Creature* spectator : spectators) {
-		spectator->onUpdateTileItem(this, cylinderMapPos, oldItem, oldType, newItem, newType);
+	for (const auto& spectator : spectators) {
+		spectator->onUpdateTileItem(getTile(), thingMapPos, oldItem, oldType, newItem, newType);
 	}
 }
 
-void Tile::onRemoveTileItem(const SpectatorVec& spectators, const std::vector<int32_t>& oldStackPosVector, Item* item)
+void Tile::onRemoveTileItem(const SpectatorVec& spectators, const std::vector<int32_t>& oldStackPosVector,
+                            std::shared_ptr<Item> item)
 {
 	if (item->hasProperty(CONST_PROP_MOVEABLE) || item->getContainer()) {
-		auto it = g_game.browseFields.find(this);
+		auto it = g_game.browseFields.find(getTile());
 		if (it != g_game.browseFields.end()) {
 			it->second->removeThing(item, item->getItemCount());
 		}
 	}
 
 	resetTileFlags(item);
 
-	const Position& cylinderMapPos = getPosition();
+	const Position& thingMapPos = getPosition();
 	const ItemType& iType = Item::items[item->getID()];
 
 	// send to client
 	size_t i = 0;
-	for (Creature* spectator : spectators) {
-		if (Player* tmpPlayer = spectator->getPlayer()) {
-			tmpPlayer->sendRemoveTileThing(cylinderMapPos, oldStackPosVector[i++]);
+	for (const auto& spectator : spectators) {
+		if (auto tmpPlayer = spectator->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
		if (const auto tmpPlayer = spectator->getPlayer()) {
```